pandas/core/indexes/datetimes.py

```diff
@@ -220,24 +246,37 @@ class DatetimeIndex(DatetimeTimedeltaMixin):
     tz: Optional[tzinfo]
 
     # --------------------------------------------------------------------
-    # methods that dispatch to array and wrap result in DatetimeIndex
+    # methods that dispatch to DatetimeArray and wrap result
+
+    @dispatch_to_array_and_wrap(DatetimeArray)
+    def strftime(self, date_format) -> Index:
+        pass
 
-    @doc(DatetimeArray.tz_convert)
+    @dispatch_to_array_and_wrap(DatetimeArray)
     def tz_convert(self, tz) -> "DatetimeIndex":
-        arr = self._data.tz_convert(tz)
-        return type(self)._simple_new(arr, name=self.name)
+        pass
 
-    @doc(DatetimeArray.tz_localize)
+    @dispatch_to_array_and_wrap(DatetimeArray)
     def tz_localize(
         self, tz, ambiguous="raise", nonexistent="raise"
     ) -> "DatetimeIndex":
-        arr = self._data.tz_localize(tz, ambiguous, nonexistent)
-        return type(self)._simple_new(arr, name=self.name)
+        pass
+
+    @dispatch_to_array_and_wrap(DatetimeArray)
+    def to_period(self, freq=None) -> "PeriodIndex":
+        pass
+
+    @dispatch_to_array_and_wrap(DatetimeArray)
+    def to_perioddelta(self, freq) -> "TimedeltaIndex":
+        pass
+
+    @dispatch_to_array_and_wrap(DatetimeArray)
+    def to_julian_date(self) -> Index:
+        pass
 
-    @doc(DatetimeArray.to_period)
-    def to_period(self, freq=None) -> "DatetimeIndex":
-        arr = self._data.to_period(freq)
-        return type(self)._simple_new(arr, name=self.name)
+    @dispatch_to_array_and_wrap(DatetimeArray)
+    def isocalendar(self) -> "DataFrame":
+        pass
```

Comment:
i find the 4 line version clearer than the 3 line version

Reply:
I am happy with the existing static code, if others are happy with the repetition/duplications